core/src/main/java/com/linecorp/armeria/client/metric/ClientMetrics.java

```diff
@@ -0,0 +1,215 @@
+/*
+ * Copyright 2025 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.client.metric;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.concurrent.atomic.LongAdder;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.linecorp.armeria.client.endpoint.EndpointGroup;
+import com.linecorp.armeria.common.SessionProtocol;
+import com.linecorp.armeria.common.annotation.Nullable;
+
+/**
+ * Collects simple client-side metrics such as:
+ * <ul>
+ *     <li>The number of pending requests per {@link SessionProtocol}</li>
```

Comment:
Question) If the objective is to determine the optimal number of event loops depending on pending requests for each endpoint group/protocol, would it be enough to check the duration instead?

```
final EndpointGroup endpointGroup = ctx.endpointGroup();
ctx.log().whenComplete().thenAccept(log -> {
    final long pendingDuration = log.connectionTimings().pendingAcquisitionDurationNanos();
    final SessionProtocol sessionProtocol = log.sessionProtocol();
});
```

Reply:
I think using pending duration is also a good idea. However, adding `ClientMetrics` would be worthwhile because:
- `ClientMetrics` itself seems to provide useful information at the `ClientFactory` level.
- Using `ConnectionTimings` inside `maxNumEventLoopsFunction` method doesn't look straightforward as it may require additional processing.